source/userguide/optional_dependency.rst

```diff
@@ -0,0 +1,30 @@
+Managing Optional Dependency
+============================
+
+MindMeld depends on Duckling for numerical parsing and ElasticSearch for Entity Resolution and Question Answering. However if your application does not need these components, you can turn off them by configuring your MindMeld application.
+
+
+Turning off Duckling
+--------------------
+
+To turn off Duckling, specify an empty dictionary for the ``'system_entity_recognizer'`` key:
+
+.. code-block:: python
+
+   NLP_CONFIG = {
+       'system_entity_recognizer': {}
+   }
+
+
+Turning off ElasticSearch for Entity Resolution
+-----------------------------------------------
+
+If you choose not to use Elasticsearch, MindMeld provides a simple baseline version of entity resolution. This Exact Match Model only resolves to an object when the text exactly matches a canonical name or synonym. To use the Exact Match Model, add the following to your app config (``config.py``) located in the top level of your app folder:
```

Comment:
What happens when one has the exact match model and a populated mapping.json file?

Reply:
Not sure, @arushir what do you think?

Reply:
With the latest change, you don't need to explicitly 'turn off' ElasticSearch if you are not using the Entity Resolver component. It will not be called if there is no mapping.json

If you do want to do Entity Resolution and do not want to use ElasticSearch, there is a very simple baseline version of entity resolution, as describe in what was pasted above, and also [here](https://www.mindmeld.com/docs/userguide/entity_resolver.html#about-the-exact-match-text-similarity-model). I.e. if there you are using the exact match model and a populated mapping.json `This Exact Match Model only resolves to an object when the text exactly matches a canonical name or synonym.`

Reply:
It seems like these distinctions need to be made more explicit or reworded in this doc.